fuel-relayer/src/mock_db.rs

```diff
@@ -0,0 +1,255 @@
+use async_trait::async_trait;
+
+use std::{
+    borrow::Cow,
+    collections::{BTreeMap, HashMap},
+    sync::{Arc, Mutex},
+};
+
+use fuel_core_interfaces::{
+    common::{
+        fuel_storage::Storage,
+        fuel_tx::{Address, MessageId},
+    },
+    db::KvStoreError,
+    model::{
+        BlockHeight, ConsensusId, DaBlockHeight, Message, SealedFuelBlock, ValidatorId,
+        ValidatorStake,
+    },
+    relayer::{RelayerDb, StakingDiff, ValidatorSet},
+};
+
+#[derive(Default)]
+pub(crate) struct Data {
+    pub messages: HashMap<MessageId, Message>,
+    pub validators: HashMap<ValidatorId, (ValidatorStake, Option<ConsensusId>)>,
+    pub delegator_index: HashMap<Address, Vec<DaBlockHeight>>,
+    pub staking_diffs: BTreeMap<DaBlockHeight, StakingDiff>,
+
+    pub chain_height: BlockHeight,
+    pub sealed_blocks: HashMap<BlockHeight, Arc<SealedFuelBlock>>,
+    pub validators_height: DaBlockHeight,
+    pub finalized_da_height: DaBlockHeight,
+    pub last_committed_finalized_fuel_height: BlockHeight,
+}
+
+#[derive(Default)]
+pub(crate) struct MockDb {
+    pub data: Arc<Mutex<Data>>,
+}
+
+impl MockDb {
+    pub fn tap_sealed_blocks_mut<F, R>(&mut self, func: F) -> R
+    where
+        F: Fn(&mut HashMap<BlockHeight, Arc<SealedFuelBlock>>) -> R,
+    {
+        func(&mut self.data.lock().unwrap().sealed_blocks)
```

Comment:
I'd usually point too many `unwrap`s but since this is a MockDb, and unwraps are only assuming the mutex is not poisoned it should be fine

Reply:
Yes, it's not the most "proper" approach to data access, but we can make that assumption in tests. I do not think MockDb will be a production-ready replacement for RocksDB anytime soon, unfortunately (fortunately?).

Reply:
Within the impl of the mock db itself, using `lock()/unwrap()` internally is fine. However, I think these tests will likely need to be re-written / redesigned since they are making assertions directly on the internals of the mockdb hashmaps instead of interacting with the relayerdb interface. I think it's fine to defer fixing these tests to a broader relayer cleanup initiative though.